.github/workflows/test.yml

```diff
@@ -1,17 +1,67 @@
-name: "build-test"
-on: # rebuild any PRs and main branch changes
+on:
   pull_request:
   push:
     branches:
       - master
       - 'releases/*'
 
 jobs:
-  build: # make sure build/ci works properly
-    runs-on: ubuntu-latest
+  build_test_job:
+    name: 'Build and test job'
+    runs-on: ${{ matrix.os }}
+    strategy:
+        matrix:
+          os: [windows-latest, ubuntu-latest, macos-latest]
     steps:
-    - uses: actions/checkout@v1
-    - run: |
+
+    - name: 'Checking out repo code'
+      uses: actions/checkout@v1
+
+    - name: 'Validate build'
+      run: |
         npm install
         npm build
-        npm test
\ No newline at end of file
+    - name: 'Run L0 tests'
+      run: |
+        npm run test
+    - name : 'Run test coverage'
+      if: runner.os == 'Windows'
+      env:
+        PR_NUMBER: ${{ github.event.pull_request.number }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        $coverage_result = npm run test-coverage
+        $start = $false;
+        $middle = $false; 
+        $end = $false; 
+        $count = 0; 
+        foreach ($j in $coverage_result) 
+        { 
+          if ($j.tostring().startswith("--------------")) 
+          { 
+                if (!$start) 
+                { 
+                    $start = $true;
+                    $start_index = $count 
+                } 
```

Comment:
Did you author this script?

Reply:
Nope: Reused from 
https://dev.azure.com/mseng/AzureDevOps/_git/AzureDevOps.Docs?path=%2FPipelines%2FEnvironment%2FGitHub%20Actions%2FGitHub_Actions_CI.md&_a=preview